index.html

```diff
@@ -615,6 +577,32 @@ <h2>
             <li>Resolve <var>p</var>.
             </li>
           </ol>
+          <p class="note" title="Payment Handler Install">
+            Calling <a>set()</a> for the first time is equivalent to installing
+            a payment handler for this site.
+          </p>
+          <p class="note" title="User Consent">
+            User consent for <a>set()</a> may be determined via any of the
+            following, for example:
+            <ul>
+              <li>Accepting or rejecting a permission prompt dialog at the time
+              of <a>set()</a> call</li>
+              <li>A user agent setting to always allow or deny <a>set()</a>
+              calls from specific origins</li>
+              <li>A user agent setting to always allow or deny all calls to
+              <a>set()</a></li>
+            </ul>
+          </p>
+          <p class="note" title="User Agent Discretion">
+            A user agent may reject or accept the <a>set()</a> promise without
+            prompting the user for consent using any of the following
+            heuristics, for example:
+            <ul>
+              <li>Site engagement metrics</li>
+              <li>Safe browsing database</li>
+              <li>SSL certificate validity</li>
+            </ul>
+          </p>
```

Comment:
@rsolomakhin, I don't yet understand this part of the proposal. It seems that partly you would like to authorize browsers to do some things without user consent when bad things happen, such as detecting a malicious site or bad certificate. If that's the case, could we rephrase this as "may reject the set() promise for security reasons" (and then list some examples)?

However, I am not understanding the use cases for accepting the set() promise without user consent.

Reply:
> ot understanding the use cases for accepting the set() promise without user consent.

One use case is high site engagement stat for this user on this origin.

Reply:
Thanks, Rouslan. Let me know if the above text would work for you.

We will have to decide on whether to use MAY/SHOULD in this security considerations text.

Ian